### ISSUE TITLE
GUI

### ISSUE THREAD

**Issue by thijstriemstra** [FooDeas/raspberrypi-ua-netinst#23]
Would be nice to have a cross-platform configuration tool for this project. Something with Qt would make sense to me. I'm not sure if it can or should be backwards compatible with the original version of this fork, but I'd say it shouldn't.

Thoughts on where (the repository) it should be placed?

**Comment by thijstriemstra** [comment]
I can probably whip up something pretty easily with (Py)Qt.

**Comment by FooDeas** [comment]
Sounds nice! What can I imagine under?
A kind of ´installer-config.txt´ maker? If you'd make it open source - GitHub is a nice place to be...

**Comment by thijstriemstra** [comment]
Yeah exactly. But everytime you add a feature it also needs to be added to the GUI, which should be no problem. I think it'll add maybe 130kb to this repo so mabye its worth consdering to add it here.

**Comment by FooDeas** [comment]
With pleasure! Why don't you tell that straight? :grin:
But I'd prefer to begin in a dev branch first.

**Comment by thijstriemstra** [comment]
I think a web-based (e.g. Django) thing would be nicer and allows you to store configurations into the database etc.

**Comment by Jodes81** [comment]
Funny enough I was thinking of doing this in PHP. Hosting it online somewhere would really make it accessible. And doing it online would make it really easy for windows users so nothing like Python etc would need to be installed. The question is where it could be hosted, since regular 30MB downloads would exceed the quota for any free service I know of.

Then it could be awesome if people submit additional templates which could be selected, which, for example, configure a system with NodeJS and WiringPi, or whatever they want. A really easy interface could be made like this so beginners can install a really lean, fit-for-purpose system with absolutely minimal fuss. 

In theory, all that would be needed to get this going, for example, is a little post-install post-boot script that once basic installation is complete, clones and runs a repository designated by the template the user chose, which does all the remaining custom installation of NodeJS or whatever.

The only tricky part is the number of options that raspberrypi-ua-netinst gives makes the number of combinations large, so testing this kind of system to make sure it works for all possible settings could be a nightmare without something that automatically installs and tests the possibilities on a RPi. But at 30 minutes per install, and an awful lot of possibilities, I'm not sure how testing could be feasible. Unless someone has a supercluster of Pi's? :) Or maybe a sort of shallow-install which only undoes the actions of the post-install.txt so a new combination can be installed and tested quickly. I dunno, I don't really know what I'm doing. Most of the workings of this project are magic to me.

But either way, IMO it would make sense to make a separate repo/project for this "wizard" idea which pulls in the latest suitable release from this repo. And all this would be quite a bit of work so would be worth starting with something more basic that could be grown
